apis/Google.Cloud.Spanner.V1/Google.Cloud.Spanner.V1/SessionPoolImpl.cs

```diff
@@ -92,55 +111,71 @@ private static void LogSessionsPooled()
 
         private bool TryPop(TransactionOptions options, out SessionPoolEntry entry)
         {
-            entry = default(SessionPoolEntry);
-            //we make a reasonable attempt at obtaining a session with the given transactionoptions.
-            //but its not guaranteed.
-            lock (_sessionMruStack)
+            try
             {
-                if (_sessionMruStack.Count > 0)
+                //we make a reasonable attempt at obtaining a session with the given transactionoptions.
+                //but its not guaranteed.
+                lock (_sessionMruStack)
                 {
-                    Logger.Debug(() => "Searching for a session with matching transaction semantics.");
-                    int indexToUse = -1;
-                    for (int i = 0; i < _sessionMruStack.Count
-                        && i < MaximumLinearSearchDepth; i++)
+                    if (_sessionMruStack.Count > 0)
                     {
-                        entry = _sessionMruStack[i];
-                        if (Equals(entry.Session.GetLastUsedTransactionOptions(), options))
+                        Logger.Debug(() => "Searching for a session with matching transaction semantics.");
+                        int indexToUse = -1;
+                        for (int i = 0;
+                            i < _sessionMruStack.Count
+                            && i < MaximumLinearSearchDepth;
+                            i++)
```

Comment:
column-wise the two comparisons can fit on the same line -- that makes it easier to read.